internal/xdscache/v3/endpointstranslator.go

```diff
@@ -338,7 +346,12 @@ func equal(a, b map[string]*envoy_endpoint_v3.ClusterLoadAssignment) bool {
 func (e *EndpointsTranslator) OnAdd(obj interface{}) {
 	switch obj := obj.(type) {
 	case *v1.Endpoints:
-		e.cache.UpdateEndpoint(obj)
+		if !e.cache.UpdateEndpoint(obj) {
+			e.WithField("endpoint", k8s.NamespacedNameOf(obj)).Debug("Endpoint is not in use by a ServiceCluster, not recalculating ClusterLoadAssignments")
```

Comment:
this might be too noisy to ultimately leave in the log, but useful for testing purposes.

Reply:
Yeah, does seem like it could get very busy indeed in a busy cluster, but it definitely seems useful to be able to have this for a bit at least.

Maybe we need to consider a metric instead?

Reply:
I dropped the negative case log where the Endpoints is not relevant, since that's probably the noisiest and we can infer if there's no log that it was not relevant. I left the positive case log in at debug level.